src/axom/primal/tests/primal_clip.cpp

```diff
@@ -1954,6 +1954,121 @@ TEST(primal_clip, polygon_intersects_polygon)
   }
 }
 
+/*
+ * 3x3 grid of quads that occupies the unit square,
+ * that clip against a lower or upper triangle:
+ *
+ * +---+---+---+
+ * | \ |   |   |
+ * |  \|   |   |
+ * +---+---+---+
+ * |   | \ |   |
+ * |   |  \|   |
+ * +---+---+---+
+ * |   |   | \ |
+ * |   |   |  \|
+ * +---+---+---+
+ */
+TEST(primal_clip, polygon_intersect_robustness)
+{
+  using Polygon2D = axom::primal::Polygon<double, 2>;
+  using Point2D = axom::primal::Point<double, 2>;
+  constexpr double EPS = 1e-8;
+
+  const double x[] = {0.,
+                      1. / 3.,
+                      2. / 3,
+                      1.,
+                      0.,
+                      1. / 3.,
+                      2. / 3,
+                      1.,
+                      0.,
+                      1. / 3.,
+                      2. / 3,
+                      1.,
+                      0.,
+                      1. / 3.,
+                      2. / 3,
+                      1.};
+  const double y[] = {0.,
+                      0.,
+                      0.,
+                      0.,
+                      1. / 3.,
+                      1. / 3,
+                      1. / 3.,
+                      1. / 3.,
+                      2. / 3.,
+                      2. / 3,
+                      2. / 3.,
+                      2. / 3.,
+                      1.,
+                      1.,
+                      1.,
+                      1.};
+  const int conn[][4] = {{0, 1, 5, 4},
+                         {1, 2, 6, 5},
+                         {2, 3, 7, 6},
+                         {4, 5, 9, 8},
+                         {5, 6, 10, 9},
+                         {6, 7, 11, 10},
+                         {8, 9, 13, 12},
+                         {9, 10, 14, 13},
+                         {10, 11, 15, 14}};
+  Polygon2D fine[9];
+  for(int p = 0; p < 9; p++)
+  {
+    fine[p].addVertex(Point2D {x[conn[p][0]], y[conn[p][0]]});
+    fine[p].addVertex(Point2D {x[conn[p][1]], y[conn[p][1]]});
+    fine[p].addVertex(Point2D {x[conn[p][2]], y[conn[p][2]]});
+    fine[p].addVertex(Point2D {x[conn[p][3]], y[conn[p][3]]});
+  }
+
+  // Overlap polygons
+  Polygon2D clipLower;
+  clipLower.addVertex(Point2D {0., 0.});
+  clipLower.addVertex(Point2D {1., 0.});
+  clipLower.addVertex(Point2D {0., 1.});
+
+  Polygon2D clipUpper;
+  clipUpper.addVertex(Point2D {1., 0.});
+  clipUpper.addVertex(Point2D {1., 1.});
+  clipUpper.addVertex(Point2D {0., 1.});
+
+  // Expected VFs (volume fractions, areas) for overlaps.
+  const double lower_vf[] = {1., 1., 0.5, 1., 0.5, 0., 0.5, 0., 0.};
+  const double upper_vf[] = {0., 0., 0.5, 0., 0.5, 1., 0.5, 1., 1.};
+
+  constexpr double fine_area = (1. / 3.) * (1. / 3.);
```

Comment:
Where does this come from?

Reply:
> Where does this come from?
The arrays can be thought of as what fraction of the quad we expect the resulting area to be.
`fine_area` is to account for the fact that each quad is 1/9 the size of a unit cube. 
The area we then check for is the multiplication of the two values.

Reply:
Thanks for the explanation. Can you add this as a comment in the test?

Reply:
> Thanks for the explanation! Can you add this as a comment in the test?

Yes, added!

On a side note, I made a slight mistake with my explanation here in the PR comments.
To derive the volume fraction, we take the area of the overlapped polygon and *divide* (not multiply) it by 1/9, the original area of the quad.